test/src/main/java/ai/timefold/solver/test/api/score/stream/SingleConstraintAssertion.java

```diff
@@ -4,9 +4,55 @@
 
 import ai.timefold.solver.core.api.score.buildin.hardsoftlong.HardSoftLongScore;
 import ai.timefold.solver.core.api.score.stream.Constraint;
+import ai.timefold.solver.core.api.score.stream.ConstraintJustification;
 
 public interface SingleConstraintAssertion {
 
+    /**
+     * As defined by {@link #justifiesWith(ConstraintJustification...)}.
+     *
+     * @param justifications the expected justification.
+     * @param message sometimes null, description of the scenario being asserted
+     * @return never null
+     * @throws AssertionError when the expected penalty is not observed
+     */
+    SingleConstraintAssertion justifiesWith(String message, ConstraintJustification... justifications);
```

Comment:
I'm wondering, for the sake of consistency... the `penalizes(...)` method (and its reward counterparts) have the `message` as second argument. Maybe we deprecate those and introduce new ones where `message` goes first?

It would look silly if some methods have message first, and other have message last. This is going to need a `migration` script.